### PR TITLE
Added gotcha on mixed index availability [skip ci]

### LINKED ISSUE #896: Add 'delayed index availability' notice to direct index query documentation
I recently spent multiple hours trying to figure out why my [direct index queries](http://docs.janusgraph.org/latest/direct-index-query.html) would fail. At the end they worked all along, however the index queried did either not yet include the nodes added or was not available when the query was executed.

Adding a delay of 1s after defining the index and inserting test nodes before performing a query resolved the issue. To make this circumstance more apparent, I'd like to suggest adding another 'gotcha' to the documentation of direct index queries:

http://docs.janusgraph.org/latest/direct-index-query.html

### PR DESCRIPTION
**Pull request by chupman**
Signed-off-by: Chris Hupman <chupman@us.ibm.com>
fixes #896 

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [X] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: docs/directindex.adoc [hunk 1]
```diff
@@ -70,3 +70,7 @@ To avoid such identifier collisions, use the `setElementIdentifier` method to de
 
 [source, java]
 graph.indexQuery("vertexByText", "$v$name:v.john").setElementIdentifier("$v$").vertices()
+
+==== Mixed Index Availability Delay
+
+When a query traverses a http://docs.janusgraph.org/latest/indexes.html#index-mixed[mixed index] immediately after data is inserted the changes may not be visible. In <<elasticsearch, Elasticsearch>> the configiuration option that determines this delay is https://www.elastic.co/guide/en/elasticsearch/reference/5.4/index-modules.html#dynamic-index-settings[index refresh interval]. In <<solr, Solr>> the primary configration option is https://lucene.apache.org/solr/guide/6_6/near-real-time-searching.html[max time].
```
Comment (robertdale) [inline docs/directindex.adoc]: Use the asciidoc reference `<<index-mixed>>` instead of URLs for internal links.

**Comment by chupman** [comment]
Hi @robertdale, requested changes have been made. I wanted to ask if you want this be targeted at 2.0 and then merged to master separately. I'm not sure if squash and merge is normally use so please let me know if you want me to squash.

**Comment by robertdale** [comment]
@chupman so you can't just change the base on this PR quite yet because it wants to pull in all the master changes.  if you rebase your changes on 0.2 and force push back to this branch, then we should be able to change the base on the PR.

**Comment by chupman** [comment]
Hi @robertdale It's rebased against 2.0. 

On a side note the Travis tests were triggered even though I had [skip ci] in the commit message. Is the rule for skip ci only set for the master branch?

**Comment by sjudeng** [comment]
Looks like it's `[skip-ci]` in the commit message. Must not like the dash.

File: docs/directindex.adoc [hunk 2]
```diff
@@ -70,3 +70,7 @@ To avoid such identifier collisions, use the `setElementIdentifier` method to de
 
 [source, java]
 graph.indexQuery("vertexByText", "$v$name:v.john").setElementIdentifier("$v$").vertices()
+
+==== Mixed Index Availability Delay
+
+When a query traverses a <<index-mixed, mixed index>> immediately after data is inserted the changes may not be visible. In <<elasticsearch, Elasticsearch>> the configiuration option that determines this delay is https://www.elastic.co/guide/en/elasticsearch/reference/5.4/index-modules.html#dynamic-index-settings[index refresh interval]. In <<solr, Solr>> the primary configration option is https://lucene.apache.org/solr/guide/6_6/near-real-time-searching.html[max time].
```
Comment (pluradj) [inline docs/directindex.adoc]: s/configiuration/configuration

**Comment by chupman** [comment]
@pluradj fixed your originally noted and another distinct typo of configuration. I might need to look into adding a spell-check plugin for vim.

**Pull request merged**